File: Programação/Back-End PHP/09/Exemplo Aula/Conta.php
<?php

abstract class Conta
{
    private $agencia;
    private $titular;
    private $saldo;



    protected function saca(float $valorASacar): void
    {
        if ($valorASacar > $this->saldo) {
            echo "Saldo indisponível";
            return;
        }

        $this->saldo -= $valorASacar;
    }

    protected function depositar(float $valorADepositar): void
    {
        if ($valorADepositar < 0) {
            echo "Valor precisa ser positivo";
            return;
        }

        $this->saldo += $valorADepositar;
    }

    protected function setSaldo(float $saldo)
    {
        $this->saldo = $saldo;
    }

    protected function getSaldo(): float
    {
        return $this->agencia;
    }

    protected function setAgencia(float $agencia)
    {
        $this->agencia = $agencia;
    }

    protected function getAgencia(): float
    {
        return $this->saldo;
    }


    protected function getNomeTitular(): string
    {
        return $this->titular->recuperaNome();
    }

    protected function getCpfTitular(): string
    {
        return $this->titular->recuperaCpf();
    }

}

File: modelagem-sistemas/uml-poo/Codigos/Aula 4/Exemplo Aula/Exemplo_Conta_abstract/src/Conta.java

public abstract class Conta {
	
	private String agencia;
	private String conta;
	private float saldo;
	
	public float getSaldo() {
		return saldo;
	}
	public String getAgencia() {
		return this.agencia;
	}
	public void setAgencia(String agencia) {
		this.agencia = agencia;
	}
	public String getConta() {
		return conta;
	}
	public void setConta(String conta) {
		this.conta = conta;
	}
	
	public void depositar(float deposito) {
		saldo += deposito;
		System.out.println("Seu saldo é " + this.saldo);
	}
	public float verSaldo() {
		return this.saldo;
	}
	
}
